pkg/envoy/admin.go

```diff
@@ -26,13 +26,16 @@ import (
 
 // DrainListeners drains inbound listeners of Envoy so that inflight requests
 // can gracefully finish and even continue making outbound calls as needed.
-func DrainListeners(adminPort uint32, inboundonly bool) error {
+func DrainListeners(adminPort uint32, inboundonly bool, skipExit bool) error {
 	var drainURL string
 	if inboundonly {
 		drainURL = "drain_listeners?inboundonly&graceful"
 	} else {
 		drainURL = "drain_listeners?graceful"
 	}
+	if skipExit {
+		drainURL += "&skip_exit"
```

Comment:
Does envoy ignore this flag on older versions?

Reply:
Yes, but we currently have a tight coupling of envoy and pilot-agent anyways so it should never be used with older envoy